lib/recommender_system/irt_recommender.rb

```diff
@@ -0,0 +1,193 @@
+module RecommenderSystem
+  class IrtRecommender < RecommenderSystem::Recommender
+    # Metoda vytvori list learning objektov s priradenymi hodnotami na zaklade pravdepodobnosti spravnej odpovede urcenej cez Item Response Theory
+    def get_list
+      los = self.learning_objects
+      list = Hash.new
+      user = User.find(self.user_id)
+      proficiency = user.proficiency
+      los.each do |lo|
+        difficulty = lo.irt_difficulty || 0.5
+        discrimination = lo.irt_discrimination || 0.5
+        # Vypocet pravdepodobnosti na zaklade 2P IRT
+        probability = (Math::E**(discrimination*(proficiency-difficulty))) / (1 + Math::E**(discrimination*(proficiency-difficulty)))
+        list[lo.id] = probability
+      end
+      list
+    end
+
+    # Opakuje sa o stvrtej rano, vid. config/schedule.rb
+    def self.update_table
+      require "rinruby"
+      # Predspracovanie dat
+      result_relations = Levels::Preproces.preproces_data
+      number_of_users = User.all.order("id DESC").first.id
+      number_of_los = LearningObject.all.order("id DESC").first.id
```

Comment:
Je pravda, ze som to nenazvala najlepsie, takze to prementujem. Hm, aj ked je objekt skryty, oplati sa mi s nim ratat, pretoze moze pomoct urcit parametre pre userov. Malo by mi teda stacit neukladat jeho parametre do DB. Ked je objekt skryty a spravim `Learningobject.find(i)` vrati mi nil?